src/transaction.rs

```diff
@@ -0,0 +1,254 @@
+            let val = ffi_try!(ffi::rocksdb_transaction_get_cf(
+                self.inner,
+                readopts.inner,
+                cf.inner,
+                key.as_ptr() as *const c_char,
+                key.len() as size_t,
+                &mut val_len,
+            )) as *mut u8;
+            if val.is_null() {
+                Ok(None)
+            } else {
+                Ok(Some(DBVector::from_c(val, val_len)))
+            }
+        }
+    }
+
+    /// Read a key inside a transaction
+    ///
+    /// ReadOptions: default
+    pub fn get_cf(&self, cf: ColumnFamily, key: &[u8]) -> Result<Option<DBVector>, Error> {
+        self.get_cf_opt(cf, key, ReadOptions::default())
+    }
+
+    /// Insert a value into the database under the given key.
+    ///
+    /// ColumnFamilyHandle: default
+    pub fn put(&self, key: &[u8], value: &[u8]) -> Result<(), Error> {
+        unsafe {
+            ffi_try!(ffi::rocksdb_transaction_put(
+                self.inner,
+                key.as_ptr() as *const c_char,
+                key.len() as size_t,
+                value.as_ptr() as *const c_char,
+                value.len() as size_t,
+            ));
+            Ok(())
+        }
+    }
+
+    /// Insert a value into the database under the given key.
+    pub fn put_cf(&self, cf: ColumnFamily, key: &[u8], value: &[u8]) -> Result<(), Error> {
+        unsafe {
+            ffi_try!(ffi::rocksdb_transaction_put_cf(
+                self.inner,
+                cf.inner,
+                key.as_ptr() as *const c_char,
+                key.len() as size_t,
+                value.as_ptr() as *const c_char,
+                value.len() as size_t,
+            ));
+            Ok(())
+        }
+    }
+
+    /// Merge a key inside a transaction
+    pub fn merge(&self, key: &[u8], value: &[u8]) -> Result<(), Error> {
+        unsafe {
+            ffi_try!(ffi::rocksdb_transaction_merge(
+                self.inner,
+                key.as_ptr() as *const c_char,
+                key.len() as size_t,
+                value.as_ptr() as *const c_char,
+                value.len() as size_t,
+            ));
+            Ok(())
+        }
+    }
+
+    /// Transaction rollback
+    pub fn rollback(&self) -> Result<(), Error> {
+        unsafe { ffi_try!(ffi::rocksdb_transaction_rollback(self.inner,)) }
+        Ok(())
+    }
+
+    /// Transaction rollback to savepoint
+    pub fn rollback_to_savepoint(&self) -> Result<(), Error> {
+        unsafe { ffi_try!(ffi::rocksdb_transaction_rollback_to_savepoint(self.inner,)) }
+        Ok(())
+    }
+
+    /// Set savepoint for transaction
+    pub fn set_savepoint(&self) {
+        unsafe { ffi::rocksdb_transaction_set_savepoint(self.inner) }
+    }
+}
+
+impl<'a> Drop for Transaction<'a> {
+    fn drop(&mut self) {
+        self.snapshot = None;
+        unsafe {
+            ffi::rocksdb_transaction_destroy(self.inner);
+        }
+    }
+}
+
+impl<'a> CreateIter for Transaction<'a> {
```

Comment:
I like where this is going. I'm starting to think we should eventually move all operations into traits and into an `::ops` module, similar to `std::ops::*`. It will probably help cut down on code duplication and large files. Not something that needs to be done in this PR, but I see this going through some evolution.